crates/cdk/src/wallet/proofs.rs

```diff
@@ -166,112 +167,389 @@ impl Wallet {
+                proofs_count
+                    .entry(proof.keyset_id)
+                    .and_modify(|count| *count += 1)
+                    .or_insert(1);
+            }
+            let fee = calculate_fee(&proofs_count, &fees).unwrap_or(Amount::ZERO);
+
+            if proofs_amount >= amount + fee {
+                let proofs = paths[t as usize].clone();
+                tracing::trace!(
+                    "Selected proofs for amount {} with fee {}: {:?}",
+                    amount,
+                    fee,
+                    proofs
+                );
+                return Some(proofs);
+            }
         }
+    }
 
-        Ok(selected_proofs)
+    tracing::trace!("No proofs found for amount {}", amount);
+    None
+}
+
+/// Select proofs options
+pub struct SelectProofsOptions {
+    /// Allow inactive keys (if `true`, inactive keys will be selected first in largest order)
+    pub allow_inactive_keys: bool,
+    /// Include fees to add to the selection amount
+    pub include_fees: bool,
+    /// Proof selection method
+    pub method: ProofSelectionMethod,
+}
+
+impl SelectProofsOptions {
+    /// Create new [`SelectProofsOptions`]
+    pub fn new(
+        allow_inactive_keys: bool,
+        include_fees: bool,
+        method: ProofSelectionMethod,
+    ) -> Self {
+        Self {
+            allow_inactive_keys,
+            include_fees,
+            method,
+        }
     }
 
-    /// Select proofs to send
-    #[instrument(skip_all)]
-    pub async fn select_proofs_to_swap(
-        &self,
-        amount: Amount,
-        proofs: Proofs,
-    ) -> Result<Proofs, Error> {
-        let active_keyset_id = self.get_active_mint_keyset().await?.id;
+    /// Allow inactive keys (if `true`, inactive keys will be selected first in largest order)
+    pub fn allow_inactive_keys(mut self, allow_inactive_keys: bool) -> Self {
+        self.allow_inactive_keys = allow_inactive_keys;
+        self
+    }
 
-        let (mut active_proofs, mut inactive_proofs): (Proofs, Proofs) = proofs
-            .into_iter()
-            .partition(|p| p.keyset_id == active_keyset_id);
+    /// Include fees to add to the selection amount
+    pub fn include_fees(mut self, include_fees: bool) -> Self {
+        self.include_fees = include_fees;
+        self
+    }
 
-        let mut selected_proofs: Proofs = Vec::new();
-        inactive_proofs.sort_by(|a: &Proof, b: &Proof| b.cmp(a));
+    /// Proof selection method
+    pub fn method(mut self, method: ProofSelectionMethod) -> Self {
+        self.method = method;
+        self
+    }
+}
 
-        for inactive_proof in inactive_proofs {
-            selected_proofs.push(inactive_proof);
-            let selected_total = Amount::try_sum(selected_proofs.iter().map(|p| p.amount))?;
-            let fees = self.get_proofs_fee(&selected_proofs).await?;
+impl Default for SelectProofsOptions {
+    fn default() -> Self {
+        Self {
+            allow_inactive_keys: false,
```

Comment:
I think we want to always prefer inactive keys key. We want to encourage the behavior to rotate away from inactive keysets as quickly as possible. I'm okay with leaving it as option and not completely enforcing it like before but I think the default should be true, and maybe rename it to `prefer_inactive_keysets` to make what it controls a bit more clear. Open push back if there was a reason you defaulted to false?

Reply:
Sounds good to me!